proxy/http/HttpSM.cc

```diff
@@ -2073,7 +2073,9 @@ HttpSM::state_send_server_request_header(int event, void *data)
     free_MIOBuffer(server_entry->write_buffer);
     server_entry->write_buffer = nullptr;
     method                     = t_state.hdr_info.server_request.method_get_wksidx();
-    if (!t_state.api_server_request_body_set && method != HTTP_WKSIDX_TRACE && HttpTransact::has_request_body(&t_state, ua_txn)) {
+    if (!t_state.api_server_request_body_set && method != HTTP_WKSIDX_TRACE &&
+        ua_txn->has_request_body(t_state.hdr_info.request_content_length,
```

Comment:
Does ProxyTransaction need these two values to answer whether it has request body? I guess it already has enough info in it.

Reply:
I went back and forth on the arguments for this method.  As it stands, the Http1.x version of ProxyTransaction does not keep a parsed version of the header, so we would at least need to pass back a HTTPHdr.  Since the HttpSM has already pulled out and normalized the content_length and transfer-encoding values, it seemed easier to pass them in directly.

I think we all want to push the parsing into the ProxyTransaction class (as it already is for HTTP/2 and I assume HTTP/3).  When that happens, we can simplify this signature.

Reply:
`Http1Transaction::allow_half_open` uses `_sm->t_state`, so it is possible to remove the arguments. But I'm not sure whether accessing `t_state` from Http1Transaction is a good thing, although `t_state` is a public member.

Reply:
Hmm.  I guess these Transaction objects do have direct references to the HttpSM.  I'll open up another PR to tidy up the API's.